nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/MetricsBuilder.java

```diff
@@ -0,0 +1,128 @@
+        ProcessorId = processorId;
+        ProcessorName =processorName;
+    }
+
+    public MetricsBuilder setProcessorId(String processorId){
+        this.ProcessorId = processorId;
+        return this;
+    }
+
+    public MetricsBuilder setProcessorName(String processorName){
+        this.ProcessorName = processorName;
+        return this;
+    }
+
+    public MetricsBuilder setTags(String tags) {
+        this.Tags = tags;
+        return this;
+    }
+
+    public MetricsBuilder metric(String metricName, long count){
+        Metric metric = null;
+        if(isProcessorMetric) {
+            metric =  new Metric(this.Computer, this.ProcessGroupId, this.ProcessGroupName);
+            metric.ProcessorId = this.ProcessorId;
+            metric.ProcessorName = this.ProcessorName;
+        } else {
+            metric = new Metric(this.Computer, this.ProcessGroupId, this.ProcessGroupName);
+        }
+        metric.CategoryName = this.CategoryName;
+        metric.Name = metricName;
+        metric.setCount(count);
+        if(this.Tags != null) {
+            metric.Tags = this.Tags;
+        }
+        metrics.add(metric);
+        return this;
+    }
+
+    public MetricsBuilder metric(String metricName, double count){
+        Metric metric = null;
+        if(isProcessorMetric) {
+            metric =  new Metric(this.Computer, this.ProcessGroupId, this.ProcessGroupName);
+            metric.ProcessorId = this.ProcessorId;
+            metric.ProcessorName = this.ProcessorName;
+        } else {
+            metric = new Metric(this.Computer, this.ProcessGroupId, this.ProcessGroupName);
+        }
+        metric.CategoryName = this.CategoryName;
+        metric.Name = metricName;
+        metric.setCount(count);
+        if(this.Tags != null) {
+            metric.Tags = this.Tags;
+        }
+        metrics.add(metric);
+        return this;
+    }
+
+    public MetricsBuilder metric(String metricName, int count) {
+        Metric metric = null;
+        if(isProcessorMetric) {
+            metric =  new Metric(this.Computer, this.ProcessGroupId, this.ProcessGroupName);
+            metric.ProcessorId = this.ProcessorId;
+            metric.ProcessorName = this.ProcessorName;
+        } else {
+            metric = new Metric(this.Computer, this.ProcessGroupId, this.ProcessGroupName);
+        }
+        metric.CategoryName = this.CategoryName;
+        metric.Name = metricName;
+        metric.setCount(count);
+        if(this.Tags != null) {
+            metric.Tags = this.Tags;
+        }
+        metrics.add(metric);
+        return this;
+    }
+    public List<Metric> build() {
+        return metrics;
+    }
+
+    public String Computer;
+    public String CategoryName;
+    public String ProcessGroupId;
+    public String ProcessGroupName;
+    public String ProcessorId;
+    public String ProcessorName;
+    public boolean isProcessorMetric = false;
+    public String Tags = null;
+
```

Comment:
Same comments apply here as to the `Metric` class - we want to try to keep member variables at the top of the class declaration and stick to private member variables with appropriate getters/setters and camelCase.

Reply:
done